File: Algorithms/Problems/MaximumSubarray/tests/maximum_subarray_kadane_test.py
from Algorithms.Problems.MaximumSubarray.maximum_subarray_kadane import (
    maximum_subarray_kadane,
)


class TestMaximumSubarrayKadane:
    def test_returns_maximum_subarray(self):
        data = [1, -2, 3, 10, -5, 14]
        expected = [3, 10, -5, 14]

        result = maximum_subarray_kadane(data)

        assert result == expected

File: Algorithms/Problems/MaximumSubarray/tests/MaximumSubarrayKadane.test.ts
import { maximumSubarrayKadane } from '../MaximumSubarrayKadane';

describe('MaximumSubarray Linear', () => {
  it('should find maximum subarray', () => {
    const data = [1, -2, 3, 10, -5, 14];
    const expected = [3, 10, -5, 14];

    const result = maximumSubarrayKadane(data);

    expect(result).toEqual(expected);
  });
});
